lib/rubocop/cop/performance/flat_map.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Performance
+      # This cop is used to identify usages of
+      #
+      # @example
+      #   # bad
+      #   [1, 2, 3, 4].map { |e| [e, e] }.flatten
+      #   [1, 2, 3, 4].map { |e| [e, e] }.flatten(1)
+      #
+      #   # good
+      #   [1, 2, 3, 4].flat_map { |e| [e, e] }
+      class FlatMap < Cop
+        MSG = 'Use `flat_map` instead of `map...flatten`.'
+
+        def on_send(node)
+          receiver, second_method, flatten_param  = *node
+          return unless second_method == :flatten
+          flatten_level, = *flatten_param
+          return unless flatten_level.nil? || flatten_level == 1
```

Comment:
`flatten` and `flatten(1)` behave differently - one will flatten the array recursively and the other will flatten just one level in it. For structures with just one level of nesting there's no difference, but there will be a difference otherwise.